docs/getting_started_rules.md

```diff
@@ -0,0 +1,200 @@
+    def setUp(self):
+        """Setup"""
+        super(TestMyNewRule, self).setUp()
+        self.collection.register(MyNewRule())
+
+    def atest_file_positive(self):
+        """Test Positive"""
+        self.helper_file_positive() # By default, a set of "correct" templates are checked
+
+    def test_file_negative(self):
+        """Test failure"""
+        self.helper_file_negative('templates/bad/mynewrule.yaml', 1) # Amount of expected matches
+```
+
+#### Running the Tests
+The unit tests (and `pylint` for linting the Python code) are processed by [`Tox`](https://tox.readthedocs.io/en/latest/).
+
+Make sure Tox is installed and then run Tox by just calling it:
+
+```bash
+# Install Tox
+$ pip install tox
+
+# Run all test suites. Same command as used when validating a Pull Request
+$ tox
+
+# Run a specific test suite:
+$ tox -e py36 # Run all unit tests again Python 3.6
+```
+Tox test suites available:
+* **py27**: Unit tests (Python 2.7)
+* **pylint27**: Python syntax check (Python 2.7)
+* **py36**: Unit tests (Python 3.6)
+* **pylint36**: Python syntax check (Python 3.6)
+
+## Code Snippets
+The skeleton code mention in the [Create new Rule](#create_new_rule) step contains no real code to keep it as simple as possible. This section contains a few simple snippets to get you started. For more complex solutions, take a deep-dive in the code of the [existing rules](/src/cfnlint/rules)!
+
+### Add a RuleMatch
+
+The following snippet is a simple example on how to return a finding for every resource in the template:
+
+```python
+  # Get all resources
+  resources = cfn.get_resources()
+
+  # Loop over all the found resources
+  for resource_name, resource in resources.items():
+
+      # The path is used to determine the line/column number when returning the Match information.
+      path = ['Resources', resource_name]
+      # The error message with specific error information
+      message = 'An error message about resource {0}'
+      matches.append(RuleMatch(path, message.format(resource_name)))
+```
+
+### Check specific resource types
+The following snippet is a simple example on checking a specific resource type:
+
+```python
+  # Get all EC2 instances from the template
+  resources = cfn.get_resources(['AWS::EC2::Instance'])
+
+  # Loop over all the found resources
+  for resource_name, resource in resources.items():
+      # Check the InstanceType setting
+      properties = resource.get('Properties')
+
+      # Only check what we need. Other rules handle misconfigurations
+      if properties:
+          # Check the instance type specified
+          if properties.get('InstanceType') == 't2.small':
+
+            # The path is used to determine the line/column number when returning the Match information.
+            path = ['Resources', resource_name, 'Properties', 'InstanceType']
+            message = 'Please do not use t2.small'
+            matches.append(RuleMatch(path, message.format(resource_name)))
+```
```

Comment:
Maybe you could add some links to already merged pull requests that add a single or multiple rules?

Reply:
That's stated in the introduction of the Snippets section.

Reply:
Suggest an explicit reference to using/providing Fixtures at the end of the test case starting-point, something like

As you can see `test_file_negative()` in this unit test makes specific use of a CloudFormation template from the Fixtures folder.  It's important to provide examples of templates which _pass_ your new rule, and also templates which generate the expected _warning/error.

Reply:
Goop point. Added